src/core/records-grant-authorization.ts

```diff
@@ -1,27 +1,46 @@
 import type { MessageStore } from '../types/message-store.js';
-import type { RecordsDelete } from '../interfaces/records-delete.js';
-import type { RecordsPermissionScope } from '../types/permissions-types.js';
 import type { RecordsRead } from '../interfaces/records-read.js';
 import type { RecordsWrite } from '../interfaces/records-write.js';
+import type { PermissionsGrantMessage, RecordsPermissionScope } from '../types/permissions-types.js';
 
 import { GrantAuthorization } from './grant-authorization.js';
 import { DwnError, DwnErrorCode } from './dwn-error.js';
 
 export class RecordsGrantAuthorization {
+  public static async authorizeWrite(
+    tenant: string,
+    incomingMessage: RecordsWrite,
+    author: string,
+    messageStore: MessageStore,
+  ): Promise<void> {
+    const permissionsGrantMessage = await GrantAuthorization.authorizeGenericMessage(tenant, incomingMessage, author, messageStore);
+
+    RecordsGrantAuthorization.authorizeScope(incomingMessage, permissionsGrantMessage);
```

Comment:
Nit: can we rename to align with `verifyGrantActive()` and `verifyGrantScopeInterfaceAndMethod()`

I also want to rename `authorizeGenericMessage()` but can't think of a good name that covers all that occurs... so fine to leave as is for now.
```suggestion
    RecordsGrantAuthorization.verifyScope(incomingMessage, permissionsGrantMessage);
```